### DISCUSSION TITLE
Error with RevoGrid angular-datagrid in Standalone Components

### DISCUSSION THREAD

**Discussion by Thoemmy** [revolist/revogrid#469] [General]
**Describe the issue**
When i import the RevoGrid into my standalone component i receive following error:
```
TypeError: (0 , tslib__WEBPACK_IMPORTED_MODULE_3__.__decorate) is not a function
    at 4225 (revolist-angular-datagrid.mjs:75:22)
    at __webpack_require__ (bootstrap:19:1)
    at 6218 (IAuth.ts:18:2)
    at __webpack_require__ (bootstrap:19:1)
    at 5911 (app.navigation.ts:71:2)
    at __webpack_require__ (bootstrap:19:1)
    at 2747 (app.component.ts:159:39425)
    at __webpack_require__ (bootstrap:19:1)
    at 2927 (environment.ts:27:1)
    at __webpack_require__ (bootstrap:19:1)
    at __webpack_exec__ (.*\.entry\.js.*$ include: \.entry\.js$ exclude: \.system\.entry\.js$ strict namespace object:13:1)
    at .*\.entry\.js.*$ include: \.entry\.js$ exclude: \.system\.entry\.js$ strict namespace object:13:1
    at __webpack_require__.O (chunk loaded:23:1)
    at .*\.entry\.js.*$ include: \.entry\.js$ exclude: \.system\.entry\.js$ strict namespace object:13:1
    at webpackJsonpCallback (jsonp chunk loading:71:1)
    at main.js:1:69
```
I'm using Angular 17.3.10 and tslib 2.3.0

Inside the component i use it as following:
```typescript
import { Component } from '@angular/core';
import { RevoGrid } from '@revolist/angular-datagrid';

@Component({
  selector: 'test',
  imports: [RevoGrid],
  template: `<revo-grid [source]="rows" [columns]="columns" theme="material"></revo-grid>`,
  styleUrls: ['test.component.css'],
  standalone: true,
})
export class TestComponent {
  columns = [{ name: 'Greeting', prop: 'name' }, { prop: 'details' }];
  rows = [
    { name: 'I am', details: 'Angular' },
    { name: 'Hello', details: 'Angular' },
  ];
}

```

**Comment by revolist** [comment]
hello @Thoemmy 
could you please reproduce it with this configuration?
https://codesandbox.io/p/sandbox/rg-start-angular-standalone-pz6lv9?file=%2Fsrc%2Fapp%2Fapp.component.ts

Or please convert it to a devbox (or make a new one) on codesandbox.io, this gives you isolated env for better testing.

Thank you for reaching us here.

**Comment by Thoemmy** [comment]
Ive setted up a devbox but still getting errors, but this time different errors
[sandbox](https://codesandbox.io/p/devbox/x5h9kv?migrateFrom=slsh8q&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyrfkejr00063j6npm370hh6%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyrfkejr00023j6n6tobh5po%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyrfkejr00043j6nyeeas9z9%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyrfkejr00053j6n92tr24ej%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyrfkejr00023j6n6tobh5po%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyrfkejr00013j6n3czv8kuv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%257D%255D%252C%2522id%2522%253A%2522clyrfkejr00023j6n6tobh5po%2522%252C%2522activeTabId%2522%253A%2522clyrfkejr00013j6n3czv8kuv%2522%257D%252C%2522clyrfkejr00053j6n92tr24ej%2522%253A%257B%2522id%2522%253A%2522clyrfkejr00053j6n92tr24ej%2522%252C%2522activeTabId%2522%253A%2522clyrfqapv00bz3j6ns3dp1o3g%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A8080%252C%2522id%2522%253A%2522clyrfq2qe00ay3j6nt60a3lff%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522VIRTUAL_MACHINE%2522%252C%2522id%2522%253A%2522clyrfqapv00bz3j6ns3dp1o3g%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clyrfkejr00043j6nyeeas9z9%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyrfkejr00033j6nri3qtd5z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clyrfkex1000adbf87qqt0jal%2522%257D%255D%252C%2522id%2522%253A%2522clyrfkejr00043j6nyeeas9z9%2522%252C%2522activeTabId%2522%253A%2522clyrfkejr00033j6nri3qtd5z%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

If made from your sandbox a devbox and added all my deps and devdeps

**Comment by Thoemmy** [comment]
I found the issue!!

I am using nx (19.0.7) monorepo with angular 17.3.10
In the root tsconfig.base.json there was in the paths following line (by default) which i removed

```
    "paths": {
      "tslib": ["node_modules/tslib/tslib.d.ts"] // remove this line and revogrid works
    }
```

**Comment by Thoemmy** [comment]
> Ive setted up a devbox but still getting errors, but this time different errors [sandbox](https://codesandbox.io/p/devbox/x5h9kv?migrateFrom=slsh8q&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clyrfkejr00063j6npm370hh6%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clyrfkejr00023j6n6tobh5po%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clyrfkejr00043j6nyeeas9z9%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clyrfkejr00053j6n92tr24ej%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clyrfkejr00023j6n6tobh5po%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyrfkejr00013j6n3czv8kuv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%257D%255D%252C%2522id%2522%253A%2522clyrfkejr00023j6n6tobh5po%2522%252C%2522activeTabId%2522%253A%2522clyrfkejr00013j6n3czv8kuv%2522%257D%252C%2522clyrfkejr00053j6n92tr24ej%2522%253A%257B%2522id%2522%253A%2522clyrfkejr00053j6n92tr24ej%2522%252C%2522activeTabId%2522%253A%2522clyrfqapv00bz3j6ns3dp1o3g%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A8080%252C%2522id%2522%253A%2522clyrfq2qe00ay3j6nt60a3lff%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%2522%257D%252C%257B%2522type%2522%253A%2522VIRTUAL_MACHINE%2522%252C%2522id%2522%253A%2522clyrfqapv00bz3j6ns3dp1o3g%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clyrfkejr00043j6nyeeas9z9%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clyrfkejr00033j6nri3qtd5z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clyrfkex1000adbf87qqt0jal%2522%257D%255D%252C%2522id%2522%253A%2522clyrfkejr00043j6nyeeas9z9%2522%252C%2522activeTabId%2522%253A%2522clyrfkejr00033j6nri3qtd5z%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
> 
> If made from your sandbox a devbox and added all my deps and devdeps

Maybe you want to check this issue too

**Comment by m2a2x** [comment]
@Thoemmy are you getting errors locally different from the one in devbox?
In devbox seems there is a dependency problem, somehow it's not installing dependent library.

I've checked multiple angular installations from scratch and it seems to be working totally fine for me with revogrid.
Also seems tslib is getting exported as dependency automatically by angular which but it's not causing any problems on my side.

**Reply by Thoemmy** [reply]
On local build everything is now working, as i mentioned i removed the path to tslib in the tsconfig.base.json

in dev box i took your example and added the deps like i have in my local project (cause i wanted to see if some lib is causing the error), it could be that sandbox has some problems, idk

**Reply by m2a2x** [reply]
Got it, probably we'll need devbox for angular too. Sandbox is working ok for now.

**Reply by Thoemmy** [reply]
Okay so in sandbox i can use the sample

**Comment by Thoemmy** [comment]
Is there a full sample in Angular which you could provide?
Cause i tried several things to get it like the main demo but i was not able to reach it.
Probably there are some bugs in there.

If its ok then i want to help you to improve the angular datagrid tabel

**Comment by revolist** [comment]
Hey @Thoemmy
sure, have you tried https://rv-grid.com/guide/angular/ these examples?
What kind of sample features do you need? 

We could quickly make one for you

**Reply by Thoemmy** [reply]
Ok this site is new ro me but its the same example i received from @m2a2x i guess in the sandbox.

Because when you search for revo grid in google for example you do not land on this page.

It would be really great if you could make an example with the filter and with cell editing (excel like).

I added the property [filter]="true" in the revo-grid html elememnt but nothing was displayed.

Edit: i will look through this new site xD i think there i will find a lot more information than on the old github.io page which only has ngmodules of angular

**Reply by revolist** [reply]
We have updated the old site with deprecation notice and links to the new site. Old one was more visited hence it is getting on top of the search atm.

**Reply by Thoemmy** [reply]
Thats really great :) i saw it now on the old site.

**Reply by revolist** [reply]
@Thoemmy 
Made you an example with filter. It's appeared there was an issue in angular build which was skipping auto defining filter panel. Fixed it recently in 4.8.10+

https://codesandbox.io/p/devbox/rg-example-angular-standalone-xtkjwj

**Reply by Thoemmy** [reply]
I will test it tomorrow and will give u a feedback, thank you a lot for your help in advance and for fixing the issue

**Reply by revolist** [reply]
Ty for helping to identify it!

**Reply by Thoemmy** [reply]
Ty for fixing the issue :)
And its also a pleasure to help ^^

I tested the new version and its not working as expected, thank you a lot :)
